### PR TITLE
Migrate from requests to axios

### PR DESCRIPTION
**Pull request by jesec**
Per suggestion of https://github.com/vercel/pkg-fetch/pull/144#pullrequestreview-625308160.

### CODE REVIEW

File: lib/requests.ts [hunk 1]
```diff
@@ -13,55 +14,34 @@ export async function downloadUrl(url: string, file: string) {
...
-        p = state.size.transferred / state.size.total;
+      data.on('data', (chunk: string) => {
+        if (totalSize != null && totalSize !== 0) {
+          currentSize += chunk.length;
+          log.showProgress((currentSize / totalSize) * 100);
+        }
+      });
+      data.pipe(ws);
+
+      return promisify(stream.finished)(ws).then(() => {
```
Comment (robertsLando) [inline lib/requests.ts:33]: Not 100% sure but maybe this requires that the returned value is binded to stream? 
```suggestion
      return promisify(stream.finished).call(stream, ws).then(() => {
```

File: lib/index.ts [hunk 2]
```diff
@@ -26,6 +27,7 @@ async function download(
 
   try {
     await downloadUrl(url, local);
+    await plusx(local);
```
Comment (robertsLando) [inline lib/index.ts:30]: What is this?

File: lib/requests.ts (hunk 1 above)
Comment (jesec) [reply lib/requests.ts:33]: Not really. I tested locally. You can use a hardcoded URL in https://github.com/vercel/pkg-fetch/blob/ba011e447459891e1c7841c4a600aa8488098364/lib/index.ts#L25. For instance, `https://github.com/jesec/pkg-fetch/releases/download/v2.7/uploaded-v2.7-node-v14.16.0-linux-x64`. 

to test yourself.

File: lib/index.ts (hunk 2 above)
Comment (jesec) [reply lib/index.ts:30]: chmod +x . downloaded files don't have `x` perm by default.

Comment (leerob) [reply lib/index.ts:30]: My question - why is it needed if it wasn't there before? We're just switching the request library, right?

Comment (jesec) [reply lib/index.ts:30]: Oh. It was actually there. I forgot to add this when I simplify functions in https://github.com/vercel/pkg-fetch/commit/e314198e538d55d0eafba15e064ca027e584d313 .

**Comment by robertsLando** [comment]
@leerob @jesec If it's not clear I'm ok to release a 3.0.0 but I think we firstly need to end this: #135

**Comment by jesec** [comment]
I don't think we should release before we got the binaries ready. But we can bump the version.

**Comment by robertsLando** [comment]
Yeah maybe I explained wrongly I don't want to release it now, I have updated my comment

**Comment by jesec** [comment]
I can take over the workflow branch, if you don't mind. To start building binaries, we have to be able to pull RHEL tools, per discussions in #124. I have sent necessary info to @leerob . 

Additionally, we have to add cross-compiling support (fix `Error! Not able to build for 'arm64' here, only for 'x64'`). Github Actions don't have `arm64` host and emulation is soooo slow.

**Comment by robertsLando** [comment]
What about using travis for arm64?

**Comment by leerob** [comment]
Yes, I will cut a release after we get CI working 👍

**Comment by jesec** [comment]
Thanks. 

I like Github Actions more. I think it could be beneficial to have support for cross-compiling in this project, as well. But yeah, if we do need native, we can use Travis or Azure.

**Comment by robertsLando** [comment]
> I like Github Actions more

Me too but AFAIK there is no arm64 suppport in available runners yet. We could ask @leerob if @vercel can provide us a custom arm64 runner for this purpose

**Comment by leerob** [comment]
Would like to stick with GitHub Actions if at all possible. Could we enumerate the pros/cons in the CI PR?

**Comment by leerob** [comment]
If it's possible to get a custom arm runner, let me know and send some docs and I can ask!

**Comment by robertsLando** [comment]
@leerob Here you are: https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners

**Comment by leerob** [comment]
Let's start with getting GH Actions CI working and then I will explore this! Thank you!

**Pull request merged**